### PR TITLE
Enable ETW EventSource Tests

### LINKED ISSUE #27147: BasicEventSourceTests failing in non elevated Windows runs
cc @brianrob 

Not all our Windows test runs are elevated. Tests need to pass non elevated.

```
BasicEventSourceTests.FuzzyTests.Test_Write_Fuzzy | 35 ms | 10
-- | -- | --
BasicEventSourceTests.TestsWriteEvent.Test_WriteEvent_ComplexData_SelfDescribing_ETW | 0 ms | 10
BasicEventSourceTests.TestsWriteEvent.Test_WriteEvent_SelfDescribing_ETW | 0 ms | 10
BasicEventSourceTests.TestsWriteEvent.Test_WriteEvent_Manifest_ETW | 0 ms | 10
BasicEventSourceTests.TestsWriteEvent.Test_WriteEvent_ByteArray_Manifest_ETW | 0 ms | 10
BasicEventSourceTests.TestsWriteEvent.Test_WriteEvent_ByteArray_SelfDescribing_ETW | 0 ms | 10
BasicEventSourceTests.TestsWrite.Test_Write_T_ETW | 0 ms | 10
BasicEventSourceTests.TestsUserErrors.Test_BadEventSource_MismatchedIds
```
eg 
https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_debug/lastCompletedBuild/testReport/BasicEventSourceTests/FuzzyTests/Test_Write_Fuzzy/

They are all failing this way
```
MESSAGE:
System.Exception : Need to be elevated to run.
+++++++++++++++++++
STACK TRACE:
at BasicEventSourceTests.EtwListener..ctor(String dataFileName, String sessionName) in D:\j\workspace\windows_nt_debug88592b53\src\System.Diagnostics.Tracing\tests\BasicEventSourceTest\Harness\Listeners.cs:line 170 at BasicEventSourceTests.FuzzyTests.Test_Write_Fuzzy() in D:\j\workspace\windows_nt_debug88592b53\src\System.Diagnostics.Tracing\tests\BasicEventSourceTest\FuzzyTests.cs:line 2751
```

Fix seems to be to change all these to have `[ConditionalFact(nameof(IsProcessElevated))]`

### PR DESCRIPTION
**Pull request by brianrob**
- Upgrade TraceEvent to 2.0.5, xunit.performance to 1.0.0-beta-build0017, and CommandLineParser to 2.2.1.  This keeps the dependencies between CoreFx and xunit.performance consistent.
 - Skip ETW tests in non-elevated processes.
 - Enable ETW tests.

Fixes #27147.

### CODE REVIEW

**Comment by vancem** [comment]
Generally looks good to me.   (although there are test failures at this time).  

The other thing that worries me is that the tests simply do nothing if the process is not elevated.   If there is a way of make a warning that would make me feel better.  If not we just live with it.

**Comment by brianrob** [comment]
@vancem, yup, the test failures were a mismatched nuget dependency between CoreFx and XUnit-Performance.  This should be fixed now.

If the process is not elevated, you will get a warning in the logs with a listing of each test that is disabled.

**Comment by brianrob** [comment]
@dotnet-bot help

**Comment by brianrob** [comment]
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop Windows x64 Release Build
@dotnet-bot test Windows_NT perf

**Comment by brianrob** [comment]
I am currently working through these failures.  They are mostly about conflicting dependencies.

File: src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsUserErrors.cs [hunk 1]
```diff
@@ -76,33 +76,37 @@ public void Test_BadEventSource_MismatchedIds()
...
-            
-                    var listenerGenerators = new Func<Listener>[]
-                    {
-                        () => new EventListenerListener(),
+
+            var listenerGenerators = new List<Func<Listener>>();
+            listenerGenerators.Add(() => new EventListenerListener());
 #if USE_ETW
-                        () => new EtwListener()
+            if(TestUtilities.IsProcessElevated)
```
Comment (stephentoub) [inline src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsUserErrors.cs]: Nit: we generally use a space btw if and (

File: src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsUserErrors.cs [hunk 2]
```diff
@@ -76,33 +76,37 @@ public void Test_BadEventSource_MismatchedIds()
...
 #endif // USE_ETW
-                    };
 
-                    var settings = new EventSourceSettings[] { EventSourceSettings.Default, EventSourceSettings.EtwSelfDescribingEventFormat };
+            var settings = new EventSourceSettings[] { EventSourceSettings.Default, EventSourceSettings.EtwSelfDescribingEventFormat };
 
-                    // For every interesting combination, run the test and see that we get a nice failure message.  
-                    foreach (bool onStartup in onStartups)
+            // For every interesting combination, run the test and see that we get a nice failure message.  
+            foreach (bool onStartup in onStartups)
```
Comment (stephentoub) [inline src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsUserErrors.cs]: Would it make sense to do this as a Theory?

File: src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsWrite.cs [hunk 3]
```diff
@@ -454,6 +463,15 @@ public void Test_Write_T_In_Manifest_Serialization()
                         Assert.Equal("hi", (string)_event.PayloadValue(1, "arg2"));
                     }
                 }
+                finally
+                {
+#if USE_ETW
+                    if(etwListener != null)
+                    {
+                        etwListener.Dispose();
+                    }
```
Comment (stephentoub) [inline src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsWrite.cs]: Nit: etwListener?.Dispose();

**Comment by brianrob** [comment]
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop Windows x64 Release Build
@dotnet-bot test Windows_NT perf

**Comment by brianrob** [comment]
@dotnet-bot test Windows x86 Release Build (Unrelated System.Net.Http.Tests failure).

**Comment by brianrob** [comment]
@dotnet-bot test Windows x64 Debug Build (Unrelated System.Net.Http.Tests failure).

**Comment by brianrob** [comment]
Inner Loop: Pass
Outer Loop: 2 intermittent failures
Perf: 1 unrelated failure (failing in official runs)

**Pull request merged**